src/literacy.rs

```diff
@@ -3,19 +3,15 @@ compile_error!("feature \"std\" and \"use-core2\" cannot be enabled together.");
 
 #[derive(Debug)]
 pub enum Error {
-    #[cfg(feature = "std")]
-    Std(::std::io::Error),
-
-    #[cfg(feature = "use-core2")]
-    Core2(core2::io::Error),
+    Wrapped(::alloc::boxed::Box<dyn Marker>),
 
     // Needed for write_all blanket implementation
     WriteZero,
     Interrupted,
-
-    Other,
 }
 
+pub trait Marker: ::core::fmt::Display + ::core::fmt::Debug {}
```

Comment:
We should definitely include [`Any`](https://doc.rust-lang.org/core/any/trait.Any.html) here if possible. This would allow a user that knows which error they expect (typical use case: you use either `std` or `core2`) to downcast to that error for further introspection. Maybe we'd even want to provide a helper method for that.

Also: I hope we can come up with a better name, it's not really a marker trait imo. Maybe call it `LiteracyError`?